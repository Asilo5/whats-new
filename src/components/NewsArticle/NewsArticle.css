.news-article {
  background-color: white;
  width: 46%;
  margin: 17px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.article-button {
    background: #B56AF0;
    width: 100%;
    border: none;
    padding: 5px;
    font-size: 20px;
    color: white;
  }

  h2, p {
    margin: 10px;
  }

  img {
    width: 34em;
    height: 13em;
  }