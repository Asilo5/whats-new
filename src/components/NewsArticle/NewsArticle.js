import React from 'react';
import './NewsArticle.css';

const NewsArticle = (props) => {
    return (
        <section className='news-article'>
            <img src={props.article.img} alt={props.article.headline} />
            <h2>{props.article.headline}</h2>
            <p>{props.article.description}</p>
            <button className='article-button'>
            <a href={props.article.url}>Link To Article -></a>
            </button>
        </section>
    )
}

export default NewsArticle;