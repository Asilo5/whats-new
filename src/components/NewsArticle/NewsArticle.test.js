import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle',() => {
  it('should match the snapshot with all data passed in correctly', () => {

    const article = [{
        id: 1,
        headline: 'Spider-Man Will Remain in the Marvel Cinematic Universe',
        img: 'https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp',
        description: 'The web slinger will be the only superhero with the ability to cross between Disney and Sonyâ€™s cinematic universes, according to Marvel Studios.',
        url: 'https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html'  
    }]


    const wrapper = shallow(<NewsArticle 
              key={article.id} 
              article={article}
              />);

    expect(wrapper).toMatchSnapshot();
  })
})