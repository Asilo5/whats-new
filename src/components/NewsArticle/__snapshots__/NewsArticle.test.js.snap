// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed in correctly 1`] = `
<section
  className="news-article"
>
  <img />
  <h2 />
  <p />
  <button
    className="article-button"
  >
    <a>
      Link To Article -&gt;
    </a>
  </button>
</section>
`;
